sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -628,24 +623,59 @@ case class Substring(str: Expression, pos: Expression, len: Expression)
 
   override def eval(input: InternalRow): Any = {
     val string = str.eval(input)
-    val po = pos.eval(input)
-    val ln = len.eval(input)
-
-    if ((string == null) || (po == null) || (ln == null)) {
-      null
```

Comment:
I created a nested if in order to avoid evaluation of the 2nd or 3rd argument, if one of them is already null.